### ISSUE TITLE
Can I work around using gnome-terminal or xterm?

### ISSUE THREAD

**Issue by IntrepidPig** [microsoft/vscode-cpptools#515]
I'd rather use terminator than gnome terminal for debugging, but apparently I can't so I've been trying to work around it. I think the only options the debugger uses is --title and -x which terminator both supports. I've tried the following things:

- Making a symlink to /usr/bin/terminator as /usr/bin/gnome-terminal -> `Unable to start debugging: Input string was not in the correct format`
- Making /usr/bin/gnome-terminal a shell script that calls `terminator $@` -> Same as the previous
- Setting "externalConsole"=false

Is there anything I can do to even completely disable the external terminal all together?

**Comment by agauniyal** [comment]
@IntrepidPig  There is an option for setting external terminal in user settings of vscode(the global user settings for editor, not this plugin).

**Comment by pieandcakes** [comment]
@agauniyal that will only change the terminal in VSCodeand not the output terminal for your application. 

@IntrepidPig At this time, there isn't anything you can do to change the behavior. This is what we pass to gnome-terminal:
```
"--title DebuggerTerminal -x bash -c \"echo $$ > /tmp/Microsoft-MIEngine-fifo-zdyn1dd4.5uj; cd /src/testapp; DbgTerm=`tty`; trap 'rm /tmp/Microsoft-MIEngine-fifo-8q3yxmxv.psj /tmp/Microsoft-MIEngine-fifo-jfsdripn.7he /tmp/Microsoft-MIEngine-fifo-zdyn1dd4.5uj' EXIT; /usr/bin/gdb --interpreter=mi --tty=$DbgTerm < /tmp/Microsoft-MIEngine-fifo-8q3yxmxv.psj > /tmp/Microsoft-MIEngine-fifo-jfsdripn.7he & clear; pid=$! ; echo $pid > /tmp/Microsoft-MIEngine-fifo-zdyn1dd4.5uj; wait $pid; \""
```

the block of code is in: 

https://github.com/Microsoft/MIEngine/blob/master/src/MICore/TerminalLauncher.cs

This is an open source repo so if you would like to contribute, that would make it more likely to happen.

**Comment by ghost** [comment]
If anyone stumbling upon this is trying to use xfce4-terminal: symlinking gnome-terminal seems to work fine. Uninstall gnome-terminal and this should do the trick:

```
sudo ln -s /usr/bin/xfce4-terminal /usr/bin/gnome-terminal
```

**Comment by pieandcakes** [comment]
This should be fixed with our implementation of VS Codes `RunInTerminal` protocol. You can now specify which terminal application to use in the VS Code settings. This was released in version `vscode-cpptools version 0.20.0`
